web/src/main/java/org/cbioportal/web/ReferenceGenomeGeneController.java

```diff
@@ -0,0 +1,86 @@
+package org.cbioportal.web;
+
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiParam;
+import org.cbioportal.model.ReferenceGenomeGene;
+import org.cbioportal.service.ReferenceGenomeGeneService;
+import org.cbioportal.service.exception.GeneNotFoundException;
+import org.cbioportal.web.config.annotation.PublicApi;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RestController;
+
+import javax.validation.constraints.Size;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@PublicApi
+@RestController
+@Validated
+@Api(tags = "Reference Genome Genes", description = " ")
+public class ReferenceGenomeGeneController {
+
+    private static final int GENE_MAX_PAGE_SIZE = 100000;
+    private static final String GENE_DEFAULT_PAGE_SIZE = "100000";
+    
+    @Autowired
+    private ReferenceGenomeGeneService referenceGenomeGeneService;
+
+    @RequestMapping(value = "/reference-genome-genes/{genomeName}", method = RequestMethod.GET, produces = MediaType.APPLICATION_JSON_VALUE)
+    @ApiOperation("Get all reference genes")
+    public ResponseEntity<List<ReferenceGenomeGene>> getAllReferenceGenomeGenes(
+        @ApiParam(required = true, value = "Name of Reference Genome hg19")
+        @PathVariable String genomeName)  {
+        
+        return new ResponseEntity<>(
+                referenceGenomeGeneService.fetchAllReferenceGenomeGenes(genomeName), HttpStatus.OK);
+    }
+
+    @RequestMapping(value = "/reference-genome-genes/{genomeName}/{geneId}", method = RequestMethod.GET, produces = MediaType.APPLICATION_JSON_VALUE)
```

Comment:
This is good .. the endpoint query arguments has genomeName strings rather than referenceGenomeId (integers). Also Entrez_gene_id is good (rather than the internal genetic_entity_id). The one thing I am concerned with here is that I think we are exposing the internal id for the referenceGenome in the output from the API call. referenceGenomeId is a data member of ReferenceGenomeGene so it is being serialized in the output here. I would recommend that we inhibit the output of this auto-increment field when serializing this data in the output to the web APIs. I believe we already do that already for other data types such as Sample through the CustomObjectMapper class using the "mixin" technique. You can see in SampleMixin.java there is an annotation "@JsonIgnore" over the internalId member in the Sample model. That prevents our internal identifiers from being exposed to outside users. I think I request that we do the same thing for the referenceGenomeId in the ReferenceGenomeGene model.

Reply:
very good points! Definitely, I will add one for ReferenceGenomeGene.

Reply:
added JsonIgnore for referenceGenomeId.